### PR TITLE
feat: Adaptations to support OpenBSD port

### PR DESCRIPTION
**Pull request by VlkrS**
I understand that OpenBSD is not a supported platform, but since it would make maintaining the OpenBSD port a lot easier, I assembled a few patches that I believe should not get in your way for your consideration.

In those cases where I found the FreeBSD port to contain the same adaptations, I included FreeBSD as well.

Edit: This is part of the resolution to #8012. I would not consider the remaining patches we need to be fit for inclusion in deno mainline.

### CODE REVIEW

**Review by aapoalas — APPROVED** [review]
Changes are simple enough: I am in favour.

**Review by littledivy — APPROVED** [review]
LGTM

**Review by lino-levan — APPROVED** [review]
LGTM. Would be cool to officially support OpenBSD in the future but CI still seems to be an issue.

**Comment by VlkrS** [comment]
> LGTM. Would be cool to officially support OpenBSD in the future but CI still seems to be an issue.

If that is on option on the table, I can try to whip some of the more intrusive changes needed for the port into shape as well.

File: ext/fs/30_fs.js [hunk 1]
```diff
@@ -315,7 +315,8 @@ const { 0: statStruct, 1: statBuf } = createByteStruct({
 });
 
 function parseFileInfo(response) {
-  const unix = core.build.os === "darwin" || core.build.os === "linux";
+  const unix = core.build.os === "darwin" || core.build.os === "linux" ||
+    Deno.build.os === "freebsd" || Deno.build.os === "openbsd";
```
Comment (mmastrac) [inline ext/fs/30_fs.js:319]: `core.build.os`

**Review by mmastrac — COMMENTED** [review]
LGTM with one nit

File: ext/fs/30_fs.js (hunk 1 above)
Comment (VlkrS) [reply ext/fs/30_fs.js]: Thanks, PR updated

**Comment by VlkrS** [comment]
> LGTM with one nit

Nit fixed :-)

**Comment by VlkrS** [comment]
> > LGTM with one nit
> 
> Nit fixed :-)

Any chance that this might make it into 1.35?

**Review by aapoalas — APPROVED** [review]
LGTM

**Pull request merged**